### PR TITLE
Set member sand renewal date

### LINKED ISSUE #199: Set membership renewal date
When a new member joins and their payment is received, set their `renewal_datetime` to `now+one_year`.

Those who join on a 29 Feb renew on the following 1 Mar.

If member renews (when the next years payment clears), `renewal_datetime=renewal_datetime+one_year`

Split to #88

### LINKED ISSUE #225: Create renewal date for member upon registration
Since working on setting the sand renewal date for members, it has become apparent that the renewal date for each member should be set in the registration process - before payment. This is because a user can create an account and navigate away from the payment process allowing them not to pay.

It will help solve the following problem:
- Issue #77 shows the different actions which will be carried out after non-payment but in order to do these there must be a member renewal date. Currently there is no trivial way of checking when the membership was created, making it hard to verify.

The easy fix for this is to set for this is to set the `renewal_date` or `member_created` for the member when they complete registration so that their status can be easily monitored.

### PR DESCRIPTION
**Pull request by SamWinterhalder**
There are multiple cases for when a member makes a sand payment. 

New members can only be on-time to make a payment as there will not be a value for `renewal_date` when the payment is made. In this case the `renewal_date` is set to the `now + 1 year`.

Existing members can be:
- Early to make a payment
- Late to make a payment
- Exactly on-time to make a payment

For each of these three cases the `renewal date` should be `renewal_date + 1 year`.

Also created a test to confirm that when a new member makes their first payment, the value of `renewal_date` field is changed from `NULL` to a `datetime` object.

Fixes #199
Fixes #225

### CODE REVIEW

**Review by SamWinterhalder — COMMENTED** [review]
I love typos 😄

File: memberships/tests/test_stripe_webhooks.py [hunk 1]
```diff
@@ -114,3 +116,25 @@ def test_a_successful_payment_for_membership_gets_logged_in_db(self):
         payments = Payment.objects.all()
 
         self.assertEqual(1, payments.count())
+
+    def test_a_member_is_given_a_membership_renewal_date_upon_payment(self):
```
Comment (CarwynNelson) [inline memberships/tests/test_stripe_webhooks.py:120]: I think we should have separate test cases for a new member being given a renewal date and an existing member getting a renewal date.

For two main reasons:
1. Tests can act as a form of documentation, so it's nice to have it explicitly laid out that "new members renew a year from new" and "existing members renew a year from their previous renew date"
2. These are two separate code branches. If I delete the else from the `set_sand_renewal_date` function this test will still pass, but the functionality will be broken.

**Comment by SamWinterhalder** [comment]
@CarwynNelson I assume these conflicts are because of the merge commit earlier on?

Something somewhere seems to have messed it all up quite a lot and I am not sure how to fix this with out rewriting the majority of it when resolving the conflicts.

**Comment by CarwynNelson** [comment]
> @CarwynNelson I assume these conflicts are because of the merge commit earlier on?
> 
> Something somewhere seems to have messed it all up quite a lot and I am not sure how to fix this with out rewriting the majority of it when resolving the conflicts.

I'm not too sure if I'm honest. I might be able to have a look on Sunday if you are willing/able to wait that long.

**Comment by SamWinterhalder** [comment]
> 
> 
> > @CarwynNelson I assume these conflicts are because of the merge commit earlier on?
> > Something somewhere seems to have messed it all up quite a lot and I am not sure how to fix this with out rewriting the majority of it when resolving the conflicts.
> 
> I'm not too sure if I'm honest. I might be able to have a look on Sunday if you are willing/able to wait that long.

Much appreciated thank you.

**Review by jamesgeddes — APPROVED** [review]
@SamWinterhalder and I have been through this together and all looks well. 

Great work, Sam.

**Comment by SamWinterhalder** [comment]
Fixed #200

**Pull request merged**